### PR TITLE
Fix length calculations in Base64_SkipNewline

### PR DESCRIPTION
**Pull request by guidovranken**
ZD 12328

### CODE REVIEW

**Comment by SparkiDev** [comment]
ok to test

File: wolfcrypt/src/coding.c [hunk 1]
```diff
@@ -136,9 +136,12 @@ static WC_INLINE int Base64_SkipNewline(const byte* in, word32 *inLen, word32 *o
...
             return ASN_INPUT_E;
         }
-        curChar = in[j];
+
+        if (len) {
+            curChar = in[j];
+        }
     }
-    while (len && curChar == ' ') {
+    while (len > 1 && curChar == ' ') {
```
Comment (SparkiDev) [inline wolfcrypt/src/coding.c:144]: Still want Len to decrement to 0 when curChar is a space.
Check len is not 0 in loop before getting new curChat.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: wolfcrypt/src/coding.c (hunk 1 above)
Comment (guidovranken) [reply wolfcrypt/src/coding.c:144]: Not sure what you mean, can you please paste the desired code block?

Comment (SparkiDev) [reply wolfcrypt/src/coding.c:144]: while (len && curChar == ' ') {
    if ((--len) > 0)
        curChar = in[++j];
}

This way, the length goes to zero and an error is returned.

Comment (guidovranken) [reply wolfcrypt/src/coding.c]: Done. Thanks for the suggestion. I left out the extra parentheses for consistency with the ```if (---var ...)``` pattern elsewhere in the code.

**Review by SparkiDev — APPROVED** [review]

File: wolfcrypt/src/coding.c (hunk 1 above)
Comment (SparkiDev) [reply wolfcrypt/src/coding.c]: Looks good to me! :-)

**Pull request merged**